src/client/src/Views/EditUser.tsx

```diff
@@ -0,0 +1,163 @@
+﻿import React from "react";
+import {useUserService} from "../Services/UserService";
+import {useRolesService} from "../Services/RolesService";
+import {useSiteContext} from "../ContextProviders/SiteContextProvider";
+import {Role} from "../Types/Role";
+import {Site} from "../Types/Site";
+import {When} from "../Components/When";
+import {ValidationError} from "../Types/ValidationError";
+import { useRouter } from "next/navigation";
+
+type EditUserProps = {
+    setUserRoles: (user: string, site: string, roles: string[]) => Promise<any>
+    getRoles: () => Promise<any>
+    site: Site
+}
+
+export const EditUserCtx = () => {
+    const { setUserRoles } = useUserService();
+    const { getRoles } = useRolesService();
+    const { site } = useSiteContext();
+    return ( <EditUser setUserRoles={setUserRoles} getRoles={getRoles} site={site!} />)
+}
+export const EditUser = ({setUserRoles, getRoles, site} : EditUserProps) => {
+    const router = useRouter();
```

Comment:
This could have been pushed into the ctx component and the redirect function could have been passed in via props, that would keep the component as reusable as possible and would have made it simpler to test (no need to mock the router module)